src/main/java/com/connexta/search/query/configs/QueryManagerConfiguration.java

```diff
@@ -43,28 +42,21 @@ public DataStore dataStore(@NotNull final URL solrUrl) throws IOException {
         (SolrDataStore)
             new SolrDataStoreFactory()
                 .createDataStore(
-                    Map.of(
-                        SolrDataStoreFactory.URL.key,
-                        new URL(solrUrl, solrUrl.getPath() + "/" + SOLR_COLLECTION)));
+                    Map.of(SolrDataStoreFactory.URL.key, new URL(solrUrl, SOLR_COLLECTION)));
 
     final SolrLayerConfiguration solrLayerConfiguration =
         new SolrLayerConfiguration(new ArrayList<>());
     solrLayerConfiguration.setLayerName(SolrConfiguration.LAYER_NAME);
-    final List<SolrAttribute> layerAttributes = new ArrayList<>();
 
-    final SolrAttribute contentsSolrAttribute =
-        new SolrAttribute(SolrConfiguration.CONTENTS_ATTRIBUTE_NAME, String.class);
-    contentsSolrAttribute.setEmpty(false);
-    contentsSolrAttribute.setUse(true);
-    layerAttributes.add(contentsSolrAttribute);
+    SolrConfiguration.QUERY_TERMS.stream()
+        .forEach(
+            s -> {
+              SolrAttribute a = new SolrAttribute(s, String.class);
+              a.setEmpty(false);
+              a.setUse(true);
+              solrLayerConfiguration.getAttributes().add(a);
+            });
```

Comment:
@peterhuffer On the query side, we convert a CommonQL string to a OGC filter. An additional library is used to bridge to Solr. It accepts an OGC filter and returns a list of OGC features, which are the search results. However, we have to configure that library with every Solr field that is part of the feature type. That is done here.